src/rtc/rtc_s1_single_job.py

```diff
@@ -260,7 +260,8 @@ def compute_layover_shadow_mask(radar_grid: isce3.product.RadarGridParameters,
                                 extraiter_rdr2geo: int=10,
                                 lines_per_block_rdr2geo: int=1000,
                                 threshold_geo2rdr: float=1.0e-7,
-                                numiter_geo2rdr: int=25):
+                                numiter_geo2rdr: int=25,
+                                memory_mode: str=None):
```

Comment:
I don't think `memory_mode` can be `None` or an `str`. I think it needs to be something like: `isce3.core.GeocodeMemoryMode.SingleBlock`.

Reply:
One way to handle that, is to create a dictionary whose elements (parameters) are only added if the user provides them. Something like:
```
options_geocode = {}
if memory_mode is not None:
    options_geocode['memory_mode'] = memory_mode

geocode(...,   # list of arguments without memory_mode
                **options_geocode)
```
Notice that withe the code above, the parameter `memory_mode` is only added if it's not `None`. In that case it's safe to set the default for `memory_mode` to `None`.